From 00545dfdf3b0a74b0c9cca16e3d681f91c3f92e4 Mon Sep 17 00:00:00 2001
From: b02247 <b02247@freescale.com>
Date: Fri, 10 Aug 2012 16:35:37 +0800
Subject: [PATCH 327/635] ENGR00214857 playing some music, the sound will be
 distortion

driver change the defualt volume of the DAC, which will cause the
audio distortion. reset the DAC vol in hal

Signed-off-by: b02247 <b02247@freescale.com>
---
 alsa/config_wm8962.h | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/alsa/config_wm8962.h b/alsa/config_wm8962.h
index 12d2745..a9ac358 100644
--- a/alsa/config_wm8962.h
+++ b/alsa/config_wm8962.h
@@ -38,12 +38,17 @@
 
 #define MIXER_WM8962_DIGITAL_CAPTURE_VOLUME         "Digital Capture Volume"
 
+#define MIXER_WM8962_DIGITAL_PLAYBACK_VOLUME        "Digital Playback Volume"
 
 
 /* These are values that never change */
 static struct route_setting defaults_wm8962[] = {
     /* general */
     {
+        .ctl_name = MIXER_WM8962_DIGITAL_PLAYBACK_VOLUME,
+        .intval = 96,
+    },
+    {
         .ctl_name = NULL,
     },
 };
@@ -61,7 +66,7 @@ static struct route_setting speaker_output_wm8962[] = {
     },
     {
         .ctl_name = MIXER_WM8962_SPEAKER_VOLUME,
-        .intval = 127,
+        .intval = 121,
     },
     {
         .ctl_name = NULL,
@@ -75,7 +80,7 @@ static struct route_setting hs_output_wm8962[] = {
     },
     {
         .ctl_name = MIXER_WM8962_HEADPHONE_VOLUME,
-        .intval = 127,
+        .intval = 121,
     },
     {
         .ctl_name = NULL,
-- 
1.8.0

